homeassistant/components/zha/__init__.py

```diff
@@ -369,6 +410,104 @@ def zdo_command(self, tsn, command_id, args):
         pass
 
 
+class ZHAEvent(object):
+    """When you want signals instead of entities.
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Register callback that will be used for signals."""
+        self._hass = hass
+        self._cluster = cluster
+        self._cluster.add_listener(self)
+        self._ieee = discovery_info['endpoint'].device.ieee
+        self._ieeetail = ''.join(['%02x' % (o, ) for o in self._ieee[-4:]])
+        if discovery_info['manufacturer'] and discovery_info['model'] is not \
+                None:
+            self._id = "{}.{}_{}_{}_{}{}".format(
```

Comment:
Why is the ID so complicated ?

Reply:
I kept it what it was before to try to minimize the impact of the breaking change. Zigbee devices consist of endpoints that expose functionality on clusters as attributes. device -> endpoints -> clusters -> attributes and commands. 1 physical device may expose several functions. so what has been done up to this point is: 

domain.manufacturer_model_short_address_endpoint_id_cluster_id

here is an example of a single device that exposes motion and temperature sensors:
binary_sensor.centralite_3326l_0e35a368_1_1280
sensor.centralite_3326l_0e35a368_1_1026

does this make sense?

Reply:
yes.